aws/resource_aws_cloudwatch_event_permission.go

```diff
@@ -103,38 +103,54 @@ func resourceAwsCloudWatchEventPermissionRead(d *schema.ResourceData, meta inter
 	var policyStatement *CloudWatchEventPermissionPolicyStatement
 
 	// Especially with concurrent PutPermission calls there can be a slight delay
-	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+	var out *events.DescribeEventBusOutput
+	var err error
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
 		log.Printf("[DEBUG] Reading CloudWatch Events bus: %s", input)
-		debo, err := conn.DescribeEventBus(&input)
+		out, err = conn.DescribeEventBus(&input)
+
 		if err != nil {
 			return resource.NonRetryableError(fmt.Errorf("Reading CloudWatch Events permission '%s' failed: %s", d.Id(), err.Error()))
 		}
 
-		if debo.Policy == nil {
+		if out.Policy == nil {
 			return resource.RetryableError(&resource.NotFoundError{
 				Message: fmt.Sprintf("CloudWatch Events permission %q not found"+
 					"in given results from DescribeEventBus", d.Id()),
-				LastResponse: debo,
+				LastResponse: out,
 				LastRequest:  input,
 			})
 		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		out, err = conn.DescribeEventBus(&input)
+	}
+	if err != nil {
+		return fmt.Errorf("Reading CloudWatch Events permission '%s' failed: %s", d.Id(), err.Error())
+	}
 
-		err = json.Unmarshal([]byte(*debo.Policy), &policyDoc)
-		if err != nil {
-			return resource.NonRetryableError(fmt.Errorf("Reading CloudWatch Events permission '%s' failed: %s", d.Id(), err.Error()))
-		}
+	err = json.Unmarshal([]byte(*out.Policy), &policyDoc)
+	if err != nil {
+		return fmt.Errorf("Reading CloudWatch Events permission '%s' failed: %s", d.Id(), err.Error())
+	}
 
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
```

Comment:
Not sure how I missed this the first lookthrough, but the addition of this new `resource.Retry()` is extraneous here and will never pass after retries. The original intention of the retry logic was to treat completely a missing policy and a policy missing the new statement as the same for retries. This logic should all be contained in one big retry function as it was originally or split into its own function so the `isResourceTimeoutError()` implementation is easier. 👍